product_abc_classification/models/xyz_analysis_profile.py

```diff
@@ -0,0 +1,154 @@
+    )
+    value_criteria = fields.Selection(
+        selection=[("consumption_value", "Consumption Value")],
+        # others: 'sales revenue', 'profitability', ...
+        default="consumption_value",
+        string="Value",
+        index=True,
+        required=True,
+    )
+    past_period = fields.Integer(default=365, string="Days", required=True)
+
+    @api.depends("level_ids")
+    def _compute_representation(self):
+        def _get_sort_key_percentage(rec):
+            return rec.percentage
+
+        for profile in self:
+            profile.level_ids.sorted(key=_get_sort_key_percentage, reverse=True)
+            profile.representation = "/".join(
+                [str(x) for x in profile.level_ids.mapped("display_name")]
+            )
+
+    @api.constrains("level_ids")
+    def _check_levels(self):
+        for profile in self:
+            percentages = profile.level_ids.mapped("percentage")
+            total = sum(percentages)
+            if profile.level_ids and total != 100.0:
+                raise ValidationError(
+                    _("The sum of the percentages of the levels should be 100.")
+                )
+            if profile.level_ids and len({}.fromkeys(percentages)) != len(percentages):
+                raise ValidationError(
+                    _("The percentages of the levels must be unique.")
+                )
+
+    def write(self, vals):
+        return super().write(vals)
+
+    def _fill_initial_product_data(self, date):
+        product_list = []
+        if self.data_source == "stock_moves":
+            return self._fill_data_from_stock_moves(date, product_list)
+        else:
+            return product_list
+
+    def _fill_data_from_stock_moves(self, date, product_list):
+        self.ensure_one()
+        moves = (
+            self.env["stock.move"]
+            .sudo()
+            .search(
+                [
+                    ("sale_line_id.state", "in", ["sale", "done"]),
+                    ("sale_line_id.order_id.date_order", ">", date),
+                    ("location_dest_id.usage", "=", "customer"),
+                    ("location_id.usage", "!=", "customer"),
+                    "|",
+                    ("product_id.xyz_analysis_profile_id", "=", self.id),
+                    "|",
+                    ("product_id.categ_id.xyz_analysis_profile_id", "=", self.id),
+                    (
+                        "product_id.categ_id.parent_id.xyz_analysis_profile_id",
+                        "=",
+                        self.id,
+                    ),
+                ]
+            )
+        )
+        uom_id = self.env.ref("uom.product_uom_unit")
+        for product in moves.mapped("product_id"):
+            product_data = {
+                "product": product,
+                "units_sold": 0,
+            }
+            for move in moves.filtered(lambda m: m.product_id == product):
+                product_data["units_sold"] += move.product_uom._compute_quantity(
```

Comment:
This is not needed. The stock move already has a field for the quantity completed in the product unit of measure.